### DISCUSSION TITLE
Same average loss patterns (but different loss values) for three training runs with different learning rates

### DISCUSSION THREAD

**Discussion by BelieveDiffusion** [kohya-ss/musubi-tuner#848] [Q&A]
I'm running some fine-tunes of Qwen-Image-2512 with a custom data set of 36,000 images. As a test, I recently tried three different training runs, with the same training settings, but different learning rates. I was trying to find the "ideal" LR for this dataset in that range, or more specifically, the LR that produced the lowest overall average training loss.

To my amazement, I found that all three runs had _exactly the same average loss patterns_, just with different loss values. Here's what I mean:

<img width="1742" height="1055" alt="same_graphs" src="https://github.com/user-attachments/assets/f4c8212a-c52a-461d-904e-f2a8d191d8f7" />

The chart above shows the average loss logged by musubi-tuner over time. The learning rates are:

- Blue: 1e-5
- Green: 1e-6
- Orange: 5e-7

Vertical dotted lines indicate number of epochs completed. Orange and blue haven't trained for quite as long as green.

You can clearly see from the graph that the "sawtooth" edges of each plotline have the same pattern and timing in each runs, but the vertical scale (the average loss) differs from run to run. Sometimes one run "catches up" with another run with more of a drop in average loss, but it's still with the same basic pattern.

Is this expected? What would explain this similarity of behavior across runs? I'd assumed that earlier training would modify later learning behavior, with the learning rate being the primary driver for the difference… but the pattern is identical in each case.

Training settings:

```
"dit": "qwen_image_2512_bf16.safetensors",
"vae": "diffusion_pytorch_model.safetensors",
"text_encoder": "qwen_2.5_vl_7b.safetensors",
"dataset_config": "dataset_1024_bs2.toml",
"sdpa": True,
"mixed_precision": "bf16",
"timestep_sampling": "shift",
"weighting_scheme": None,
"discrete_flow_shift": 2.2,
"optimizer_type": "adafactor",
"fused_backward_pass": True,
"learning_rate": 1e-5, # for example
"lr_scheduler": "constant_with_warmup",
"lr_warmup_steps": 50,
"gradient_checkpointing": True,
"max_data_loader_n_workers": 2,
"persistent_data_loader_workers": True,
"max_train_epochs": 100,
"save_every_n_steps": 5000,
"seed": 42,
"output_name": "MyProject",
"output_dir": "output",
"max_grad_norm": 0.0,
```

Dataset settings:

```
[general]
resolution = [1056,1584]
caption_extension = ".txt"
batch_size = 1
enable_bucket = true
bucket_no_upscale = true

[[datasets]]
image_directory = "dataset"
cache_directory = "cache"
num_repeats = 1
```
